src/Lib/ClientOptionsValidator.php

```diff
@@ -0,0 +1,171 @@
+     *
+     * @param $options
+     *              - api_root              string|array[$mode => string]
+     *                                            API root URL to use. If you need different URLs for the
+     *                                            test and live modes, provide an array with the keys
+     *                                            'test' and 'live' and the URLs to use as values.
+     *                                            Default: "https://api.getalma.eu"
+     *
+     *              - force_tls             int|boolean 0, 1 or 2 will force TLS 1.0, 1.1 or 1.2 when connecting to the API.
+     *                                          `false` will not try to force TLS; `true` wil fallback to default value.
+     *                                          If set to 0/1/2/true, TLS will be forced even if the API ROOT uses the
+     *                                          "http://" scheme.
+     *                                          Default: 2
+     *              - mode                  string  'test' or 'live'. Default: 'live'
+     *              - logger                Psr\Log\LoggerInterface The logger instance to use for errors/warnings
+     *              - user_agent_component  array[user_agent => version] user_agent and version should be string
+     *
+     * @throws ParamsError
+     */
+    public static function validateOptions(array $options = array())
+    {
+        $config = [
+            'api_root' => [
+                Client::TEST_MODE => Client::SANDBOX_API_URL,
+                Client::LIVE_MODE => Client::LIVE_API_URL
+            ],
+            'force_tls' => 2,
+            'mode' => Client::LIVE_MODE,
+            'logger' => new NullLogger(),
+        ];
+
+        if (isset($options['api_root'])) {
+            $config['api_root'] = self::validateApiRootOption($options['api_root']);
+        }
+
+        if (isset($options['force_tls'])) {
+            $config['force_tls'] = self::validateForceTLSOption($options['force_tls']);
+        }
+
+        if (isset($options['mode'])) {
+            $config['mode'] = self::validateModeOption($options['mode']);
+        }
+
+        if (isset($options['logger'])) {
+            $config['logger'] = self::validateLoggerOption($options['logger']);
+        }
+
+        if (isset($options['user_agent_component'])) {
+            $config['user_agent_component'] = self::validateUserAgentComponentOption($options['user_agent_component']);
+        }
+        return $config;
+    }
+
+    /**
+     * @param $api_root
+     *
+     * @return array|string[]
+     * @throws ParamsError
+     */
+    public static function validateApiRootOption($api_root)
+    {
+        if (is_string($api_root)) {
+            return [
+                Client::TEST_MODE => $api_root,
+                Client::LIVE_MODE => $api_root
+            ];
+        }
+        if (isset($api_root[Client::TEST_MODE]) && isset($api_root[Client::LIVE_MODE])) {
```

Comment:
maybe we have to ensure `is_array` here (if object given, we will not be able to throw a `ParamsError`